packages/components/src/internal/components/gridbar/ExportModal.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { FC, memo, useCallback, useState } from 'react';
+import { Checkbox, Label, Modal } from 'react-bootstrap';
+import { QueryModelMap } from '../../../public/QueryModel/withQueryModels';
+
+interface ExportModalProperties {
+    queryModels: QueryModelMap;
+    tabOrder: string[];
+    onExport: (tabs: Set<string>) => Promise<void>;
+    onClose: () => void;
+    canExport: boolean;
+    title?: string;
+}
+const DEFAULT_TITLE = 'Select the Tabs to Export';
+
+export const ExportModal: FC<ExportModalProperties> = memo((props) => {
+    const {queryModels, tabOrder, onClose, onExport, canExport, title = DEFAULT_TITLE} = props;
+    const [selected, setSelected] = useState<Set<string>>(new Set(tabOrder));
```

Comment:
nit: Consider using the function initializer to avoid instantiating the new set of subsequent renders.

```tsx
const [selected, setSelected] = useState<Set<string>>(() => new Set(tabOrder));
```